### PR TITLE
[MP-1661] Separate MessageStream functionality out into its own class

### PR DESCRIPTION
**Pull request by angeloesquibel**
This PR:
Separates the MessageStream from Marigold to have its own class 
Adds the MessageStream module to Marigold Bridge

### CODE REVIEW

File: ios/RNMessageStream.m [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+@interface RNMessageStream()
+
+@property (nonatomic, strong) MARMessageStream *messageStream;
+
+@end
+
+
+@implementation RNMessageStream
+
```
Comment (igstewart3) [inline ios/RNMessageStream.m:20]: We need to add `RCT_EXPORT_MODULE();`

File: ios/RNMarigold.m [hunk 2]
```diff
@@ -59,95 +41,18 @@ + (BOOL)requiresMainQueueSetup {
     return @[@"inappnotification"];
```
Comment (igstewart3) [inline ios/RNMarigold.m:41]: Do we still need to support this event in this class?

**Review by igstewart3 — CHANGES_REQUESTED** [review]

File: ios/RNMarigold.m (hunk 2 above)
Comment (angeloesquibel) [reply ios/RNMarigold.m:41]: Hi, yes we need to add the supportedEvents. It shows me this error after building when that is not included.
Thread 1: "You must override the `supportedEvents` method of RNMessageStream"

Comment (igstewart3) [reply ios/RNMarigold.m:41]: Right, we need to implement the method but the `@"inappnotification"` event isn't emitted in this class anymore, only in the `RNMessageStream` class.

Comment (angeloesquibel) [reply ios/RNMarigold.m]: Oh right. I was looking at the wrong file. Updated the PR. Thanks Ian

**Review by igstewart3 — APPROVED** [review]
LGTM 👍

**Pull request merged**